src/tokens/mod.rs

```diff
@@ -11,24 +14,53 @@ pub struct Token {
 }
 
 impl Token {
-    fn get_match(&self) -> String {
+    pub fn get_match(&self) -> String {
         self.source[self.range.clone()].iter().collect()
     }
 
     pub fn get_name(&self) -> &Option<String> {
         &self.matcher_name
     }
+
+    pub fn first(&self) -> Option<&Token> {
+        self.children.get(0)
+    }
+
+    pub fn all_children(&self) -> Iter {
+        Iter::new(self)
+    }
+
+    pub fn named_children(&self) -> impl Iterator<Item = &Token> {
+        Iter::new(self).filter(|t| t.matcher_name.is_some())
```

Comment:
This method should return children with a specific name. If the user wants no unnamed tokens, they'd set the unnamed rule.

Reply:
I thought that was a weird method to want

Reply:
Filtering by the name of the child tokens is such a common operation that I thought it made sense to add a shorthand for it

Reply:
ya know what we should change the name to children_named, cause that sounds like what this is more than what named_children

Reply:
I think `children_by_name` might be more clear